src/transformers/modeling_utils.py

```diff
@@ -1512,6 +1513,40 @@ def _autoset_attn_implementation(
             # If a config is passed with a preset attn_implementation, we skip the automatic dispatch and use the user-provided config, with hard checks that the requested attention implementation is available.
             requested_attn_implementation = config._attn_implementation_internal
 
+        # MultiModal LLM related hack: since they consist of two or might be more sub-models
+        # we have to check and dispatch SDPA to each sub-model, in case any of them support it.
+        # If one sub-model supports SDPA while other doesn't, an error will be raised following the
+        # typical SDPA-dispatch path.
+        # Same goes for the `_supports_cache_class` because we cannot know what flag LM has
+        # before knowing which class is that LM
+        if hasattr(config, "text_config") and hasattr(config, "vision_config"):
+            text_model_cls = MODEL_MAPPING.get(type(config.text_config), None)
+            if text_model_cls is not None:
+                config.text_config._attn_implementation = config._attn_implementation
+                cls._supports_sdpa = text_model_cls._supports_sdpa
+                cls._autoset_attn_implementation(
+                    config.text_config,
+                    use_flash_attention_2=use_flash_attention_2,
+                    torch_dtype=torch_dtype,
+                    device_map=device_map,
+                    check_device_map=check_device_map,
+                )
+
+            vision_model_cls = MODEL_MAPPING.get(type(config.vision_config), None)
+            if vision_model_cls is not None:
+                config.vision_config._attn_implementation = config._attn_implementation
+                cls._supports_sdpa = vision_model_cls._supports_sdpa
+                cls._autoset_attn_implementation(
```

Comment:
Should it be this way?
```diff
-                cls._autoset_attn_implementation(
+                vision_model_cls._autoset_attn_implementation(
```

Reply:
Yes, this is actually better than the current way cause we don;t have to reset `supports_sdpa` flag every time